link_analyzer.py

```
@@ -0,0 +1,209 @@
+
+
+def analyze_single_file(file_path: str, project_root: str):
+    """Analyze a single HTML file."""
+    print(f"Analyzing file: {file_path}")
+    print("-" * 40)
+    
+    extractor = HTMLLinkExtractor(project_root)
+    results = extractor.parse_file(file_path)
+    
+    print(f"Total links: {len(results.links)}")
+    print(f"Categories found:")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)